synapse/storage/event_federation.py

```diff
@@ -398,12 +398,11 @@ def _delete_old_forward_extrem_cache_txn(txn):
             sql = ("""
                 DELETE FROM stream_ordering_to_exterm
                 WHERE
-                (
-                    SELECT max(stream_ordering) AS stream_ordering
+                room_id IN (
+                    SELECT room_id AS stream_ordering
```

Comment:
Yes, because we only want to delete entries where there are entries for the same room_id with a greater stream_ordering than the cut off point?